### PR TITLE
Add NetXDuo regression test.

### PR DESCRIPTION
**Pull request by wenhui-xie**
Add NetXDuo regression test.

### CODE REVIEW

File: test/cmake/.gitignore
Comment (bo-ms) [inline test/cmake/.gitignore:1]: do we need this file?

File: test/regression/bsd_test/bsd_test_setup.docx
Comment (bo-ms) [inline test/regression/bsd_test/bsd_test_setup.docx:1]: remove this file

File: test/cmake/.gitignore
Comment (wenhui-xie) [reply test/cmake/.gitignore:1]: The threadx and filex will be cloned to cmake folder when running the test. I added this .gitignore referring to FileX repo: https://github.com/azure-rtos/filex/blob/master/test/cmake/.gitignore.

**Review by bo-ms — APPROVED** [review]

File: test/cmake/.gitignore
Comment (wenhui-xie) [reply test/cmake/.gitignore:1]: removed.

File: test/regression/bsd_test/bsd_test_setup.docx
Comment (wenhui-xie) [reply test/regression/bsd_test/bsd_test_setup.docx:1]: removed.

**Review by TiejunMS — APPROVED** [review]

**Pull request merged**